src/main/java/com/hibernate/cinema/service/UserService.java

```diff
@@ -1,9 +1,12 @@
 package com.hibernate.cinema.service;
 
 import com.hibernate.cinema.model.User;
+import org.springframework.security.core.Authentication;
 
 public interface UserService {
     User add(User user);
 
     User findByEmail(String email);
+
+    User findUser(Authentication authentication);
```

Comment:
this should be handled on the controller layer - this is not part of the program business logic